From b3d915f5917174c7e7e57dceda3dd35e254a96de Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Sun, 6 Dec 2015 15:19:26 -0800
Subject: [PATCH 49/66] soapy: do not throw when IQ bal mode is set to off

The automatic IQ balance mode is not supported,
but we should not throw when it is set to disabled.
Setting to disabled is techinically allowable,
and currently throwing is disruptive for users.
---
 lib/soapy/soapy_source_c.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/soapy/soapy_source_c.cc b/lib/soapy/soapy_source_c.cc
index 4bd3aa3..ea84467 100644
--- a/lib/soapy/soapy_source_c.cc
+++ b/lib/soapy/soapy_source_c.cc
@@ -281,6 +281,7 @@ void soapy_source_c::set_dc_offset( const std::complex<double> &offset, size_t c
 
 void soapy_source_c::set_iq_balance_mode( int mode, size_t chan )
 {
+    if (mode == osmosdr::source::IQBalanceOff) return; //no error on disable
     throw std::runtime_error("soapy_source_c::set_iq_balance_mode() not supported");
 }
 
-- 
2.1.4

